test/e2e/e2e_coverage_and_cleanup_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var workerNodeA = "kind-worker"
+var workerNodeB = "kind-worker2"
+var cmCovDir = ".coverage/clickhouse-monitor-coverage"
+var tmCovDir = ".coverage/theia-manager-coverage"
+
+func copyCovFolder(nodeName, covDir, covPrefix string) error {
+	covDirAbs, err := filepath.Abs("../../" + covDir)
+	if err != nil {
+		return fmt.Errorf("copyCovFolder: error creating absolute file path: %v", err)
+	}
+	pathOnNode := nodeName + ":" + "/var/log/" + covPrefix + "-coverage/."
+	cmd := exec.Command("docker", "cp", pathOnNode, covDirAbs)
+	var errb bytes.Buffer
+	cmd.Stderr = &errb
+	if err := cmd.Run(); err != nil {
+		errStr := errb.String()
+		if !strings.Contains(errb.String(), "Could not find the file") {
+			return fmt.Errorf("error while running docker cp command[%v] from node: %s: %s", cmd, nodeName, errStr)
+		}
+	}
+	return nil
+}
+
+func copyCovFilesBothNodes(covPrefix string) error {
+	log.Infof("Copying coverage files from worker nodes %s and %s", workerNodeA, workerNodeB)
+	if err := copyCovFolder(workerNodeA, cmCovDir, covPrefix); err != nil {
+		return err
+	}
+	if err := copyCovFolder(workerNodeA, tmCovDir, covPrefix); err != nil {
```

Comment:
if covPrefix is "cm", you are copying clickhouse-monitor log to theia-manager folder? Am I missing something here?

Reply:
copyCovFilesBothNodes copies files from the worker nodes, which have mounted directories for clickhouse-monitor coverage and theia-manager coverage to the local .coverage directory. Within the node, these directories are /var/log/cm-coverage and /var/log/tm-coverage. Since I don't know which node the pods are running on, I copy both folders from both nodes into the corresponding local coverage directory.

For example, if both clickhouse-monitor and theia-manager were scheduled to kind-worker2, the algorithm would try to copy the contents of /var/log/cm-coverage from kind-worker, catch the error, and try again with kind-worker2. The result would be all the files from the correct node at /var/log/cm-coverage will end up in .coverage/clickhouse-monitor-coverage locally and all the files from the correct node at /var/log/tm-coverage will end up in .coverage/theia-manager-coverage